itou/www/itou_staff_views/views.py

```diff
@@ -77,20 +90,112 @@ def export_job_applications_unknown_to_ft(
         # Avoid exceedingly long filenames.
         departments_str = "multiple_departements" if len(departments) >= 5 else "_".join(departments)
-        export_ts = f"{timezone.localdate().strftime('%Y-%m-%d')}_{timezone.localtime().strftime('%H-%M-%S')}"
         writer = csv.writer(Echo())
         return StreamingHttpResponse(
             content_type="text/csv",
             headers={
                 "Content-Disposition": content_disposition_header(
                     as_attachment=True,
-                    filename=f"candidats_emplois_inclusion_{departments_str}_non_certifies_{export_ts}.csv",
+                    filename=f"candidats_emplois_inclusion_{departments_str}_non_certifies_{get_export_ts()}.csv",
                 ),
             },
             streaming_content=(writer.writerow(row) for row in content()),
         )
     return render(request, template_name, {"form": form})
+
+
+@login_required
+def export_ft_api_rejections(request):
+    if not request.user.is_superuser:
+        raise Http404
+
+    first_day_of_month = timezone.now().replace(day=1, hour=0, minute=0, second=0, microsecond=0)
+    rejected_approvals = (
+        Approval.objects.filter(
+            pe_notification_status="notification_error",
+            pe_notification_time__range=[
+                first_day_of_month - relativedelta(months=1),
+                first_day_of_month,
+            ],
+        )
+        .select_related("user", "user__jobseeker_profile")
+        .order_by("pe_notification_time")
+    )
+
+    if len(rejected_approvals) == 0:
+        messages.add_message(request, messages.WARNING, "Pas de rejets de PASS IAE sur le dernier mois")
+        return HttpResponseRedirect(reverse("dashboard:index"))
+
+    data = []
+    for approval in rejected_approvals:
+        data.append(
+            [
+                approval.number,
+                approval.pe_notification_time.isoformat(sep=" "),
+                approval.pe_notification_exit_code,
+                approval.user.jobseeker_profile.nir,
+                approval.user.jobseeker_profile.pole_emploi_id,
+                approval.user.last_name,
+                approval.user.first_name,
+                approval.user.birthdate.isoformat(),
+                approval.origin_siae_kind,
+                approval.origin_siae_siret,
+            ]
+        )
+
+    headers = [
+        "numero",
+        "date_notification",
+        "code_echec",
+        "nir",
+        "pole_emploi_id",
+        "nom_naissance",
+        "prenom",
+        "date_naissance",
+        "siae_type",
+        "siae_siret",
+    ]
+
+    workbook = generate_excel_sheet(headers, data)
+    buffer = io.BytesIO()
+    workbook.save(buffer)
+    buffer.seek(0)
+
+    return FileResponse(
+        buffer,
+        as_attachment=True,
+        filename=f"rejets_api_france_travail_{get_export_ts()}.xlsx",
+        content_type="application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+    )
```

Comment:
S’assure que `close()` sera bien appelé en cas d’exception.
```suggestion
    with io.BytesIO() as buffer:
        workbook.save(buffer)
        buffer.seek(0)

        return FileResponse(
            buffer,
            as_attachment=True,
            filename=f"rejets_api_france_travail_{get_export_ts()}.xlsx",
            content_type="application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
    )
```

Reply:
Ça ne fonctionne pas : le buffer semble se fermer trop tôt, et du coup le fichier fait 0 octets (à la vois en serveur de dev, et dans les tests)

Reply:
Effectivement. Bon, laissons comme ça, normalement ça finira bien par être garbage collecté.